src/DIG/Dialog.jsx

```diff
@@ -38,7 +38,7 @@ const Overlay = styled(`${variant}.Overlay`)`
   position: fixed;
   inset: 0;
   animation: overlayShow 150ms cubic-bezier(0.16, 1, 0.3, 1);
-  z-index: 10000;
+  z-index: 1049;
```

Comment:
This change of `z-index` and the next one on line 66 is needed to prevent this modal overlaps:

<img width="1262" alt="Знімок екрана 2023-11-13 о 16 35 22" src="https://github.com/near/near-discovery-components/assets/34593263/87bdb073-4c56-40a7-ba15-b589c98c750b">